### PR TITLE
tree model created and backbone for adding tree completed

### PR DESCRIPTION
**Pull request by karliatto**
- Tree model created
- Url for the collection added
- New backbone event to add new tree to the collection

When I try to add a new item to the collection I have the error message from the server ''' TypeError: Cannot read property 'latitude' of undefined ''' I am no sure if the server is ready to accept this post from backbone.

### CODE REVIEW

File: app/js/models/tree.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+var app = app || {};
+
+app.models = app.models || {};
+
+Backbone.Model.prototype.idAttribute = 'id';
```
Comment (chill117) [inline app/js/models/tree.js]: This overrides the idAttribute for **all** backbone models in the app. I think it's better to put this in the extend below:
```js
Backbone.Model.extend({
    idAttribute: 'id'
});
```

**Review by chill117 — CHANGES_REQUESTED** [review]
The "Add" button doesn't work for me. I am seeing the following error in the web server:
```
TypeError: Cannot read property 'latitude' of undefined
    at hobobar/server/controllers/api/tree/create.js:22:23
```

Is it working for you?

**Comment by karliatto** [comment]
- I found out we didn't have any body-parser in the server, so I added with this PR and now it's working ok, req.body was undefined.
- filename is included in the body as well  since we do not have the camera in order to be able to test it.

**Comment by chill117** [comment]
> I found out we didn't have any body-parser in the server, so I added with this PR and now it's working ok, req.body was undefined.

I don't think we need the body parser, because we are using the multer module (for file uploads).

And I don't think this will work with an actual image upload anymore. It is now just going to send a file name instead of the actual image file. It was working before [here](https://github.com/Learn-by-doing/hobobar/blob/25a6c754de2b1afeb21d2100d1ba2796ea69b4da/app/www/js/index.js#L18-L42).

It might be necessary to overwrite the tree model's `sync` method with our own custom method (using the code linked above). For example:
```js
var OurModel = Backbone.Model.extend({

	sync: function(method, model, options) {

		if (method === 'create') {

			var fileUploadOptions = new FileUploadOptions();
			fileUploadOptions.fileKey = 'image';
			// !! NOTE !!
			// `imagePath` is the fileURL  from the `navigator.camera.getPicture` function.
			fileUploadOptions.fileName = imagePath.substr(imagePath.lastIndexOf('/') + 1);
			fileUploadOptions.mimeType = 'image/jpeg';
			fileUploadOptions.params = model.toJSON();
			fileUploadOptions.chunkedMode = false;

			var fileTransfer = new FileTransfer();
			var onError = options.errors || _.noop;
			var onSuccess = options.success || _.noop;

			fileTransfer.upload(imagePath, uploadUri, function() {
				onSuccess();
			}, onError, fileUploadOptions);

		} else {
			// Do nothing.
			// We only support adding new trees.
		}
	}
});

```

**Comment by chill117** [comment]
@karliatto So what do we want to do with this? I think it's close.

**Comment by karliatto** [comment]
@chill117 I think it's close as well
by the way I wanted to share with you that I did some research in debugging cordova apps and actually chrome or chromium has a functionality for that! you go to "chrome://inspect/#devices" connect the device and open the app in the device, and you can use developers tool as in a normal website, with the current status of this PR I get an error when the app tries to find  "Request URL:file:///android_asset/www/js/leaflet-src.map". does it tell you anything?

**Comment by chill117** [comment]
@karliatto Cool! I'll give that a try :)

> I get an error when the app tries to find "Request URL:file:///android_asset/www/js/leaflet-src.map". does it tell you anything?

I don't think that error will break anything. `.map` files are for reversing JavaScript minification in the browser developer tools, to help with debugging.

File: app/index.html [hunk 2]
```diff
@@ -51,13 +58,16 @@
         </script>
 
         <script type="text/html" id="template-add-tree">
-            <label><input type="radio" name="type" value="tree">Tree</label>
-            <label><input type="radio" name="type" value="bush">Bush</label>
-            <label><input type="radio" name="type" value="herb">Herb</label>
+            <label><input id="type" type="radio" name="type" value="tree">Tree</label>
```
Comment (RobinCsl) [inline app/index.html]: The 3 inputs have the same id.

Comment (chill117) [reply app/index.html]: Good catch. I removed the duplicate IDs with a new commit.

**Pull request merged**